common/konfig.py

```diff
@@ -0,0 +1,447 @@
+
+    @ssh_proxy_port.setter
+    def ssh_proxy_port(self, value: int) -> None:
+        self['snapshots.ssh.proxy_port'] = value
+
+    @property
+    def ssh_proxy_user(self) -> str:
+        """SSH user at proxy (jump) host.
+
+        {
+            'default': 'local users name',
+            'values': 'text',
+        }
+        """
+        return self['snapshots.ssh.proxy_user']
+
+    @ssh_proxy_user.setter
+    def ssh_proxy_user(self, value: str) -> None:
+        self['snapshots.ssh.proxy_user'] = value
+
+    @property
+    def ssh_max_arg_length(self) -> int:
+        """Maximum command length of commands run on remote host. This can
+        be tested for all ssh profiles in the configuration with 'python3
+        /usr/share/backintime/common/sshMaxArg.py LENGTH'. The value '0'
+        means unlimited length.
+
+        {
+            'values': '0, >700',
+        }
+        """
+        raise NotImplementedError('see org in Config')
+        return self['snapshots.ssh.max_arg_length']
+
+    @ssh_max_arg_length.setter
+    def ssh_max_arg_length(self, length: int) -> None:
+        self['snapshots.ssh.max_arg_length'] = length
+
+    @property
+    def ssh_check_commands(self) -> bool:
+        """Check if all commands (used during takeSnapshot) work like
+        expected on the remote host.
+        { 'values': 'true|false' }
+        """
+        return self['snapshots.ssh.check_commands']
+
+    @ssh_check_commands.setter
+    def ssh_check_commands(self, value: bool) -> None:
+        self['snapshots.ssh.check_commands'] = value
+
+    @property
+    def ssh_check_ping_host(self) -> bool:
+        """Check if the remote host is available before trying to mount.
+        { 'values': 'true|false' }
+        """
+        return self['snapshots.ssh.check_ping']
+
+    @ssh_check_ping_host.setter
+    def ssh_check_ping_host(self, value: bool) -> None:
+        self['snapshots.ssh.check_ping'] = value
+
+
+class Konfig(metaclass=singleton.Singleton):
+    """Manage configuration data for Back In Time.
+
+    Dev note:
+
+        That class is a replacement for the `config.Config` class.
+    """
+
+    _DEFAULT_VALUES = {
+        'global.hash_collision': 0,
+        'global.language': '',
+        'global.use_flock': False,
+    }
+
+    _DEFAULT_SECTION = '[bit]'
+
+    def __init__(self, config_path: Path = None):
+        if not config_path:
+            xdg_config = os.environ.get('XDG_CONFIG_HOME',
+                                        os.environ['HOME'] + '/.config')
+            self._path = Path(xdg_config) / 'backintime' / 'config'
+        else:
+            self._path = config_path
+
+        logger.debug(f'Config path used: {self._path}')
+
+        self.load()
+
+        # Names and IDs of profiles
+        # EXtract all relevant lines of format 'profile*.name=*'
+        name_items = filter(
+            lambda val:
+                val[0].startswith('profile') and val[0].endswith('name'),
```

Comment:
Should it be `'.name'` rather than `'name'` to avoid picking up other keys which happen to end with `'name'`? (Hypothetical: `'profile.basename'`)
```suggestion
                val[0].startswith('profile') and val[0].endswith('.name'),
```